lib/nexthop_group.c

```diff
@@ -119,6 +125,35 @@ void nexthop_add(struct nexthop **target, struct nexthop *nexthop)
 	nexthop->prev = last;
 }
 
+void nexthop_group_add_sorted(struct nexthop_group *nhg,
+			      struct nexthop *nexthop)
+{
+	struct nexthop *position, *prev;
+
+	for (position = nhg->nexthop, prev = NULL; position;
+	     prev = position, position = position->next) {
+		if (nexthop_cmp(position, nexthop) > 0) {
```

Comment:
Looking at #4325, I see the goal is to sort all nexthops in zebra, which I think is great.

I only think we should consider using an rb-tree (with the new API) for this instead of a sorted linked-list. The choice is not too obvious in this case since nexthop lists are usually small and rb-trees have a small memory overhead associated with them (which might be significant in this case). So maybe a sorted linked-list is fine here... I'd like to hear what others think about this.

Reply:
We discussed this in the meeting a couple weeks ago and agreed this would be a good idea but should be a separate PR. When this is merge, I will make an enhancement issue to track this goal.